model/flow/sealing_segment.go

```diff
@@ -0,0 +1,193 @@
+package flow
+
+import (
+	"fmt"
+)
+
+const (
+	// expected length of a root sealing segment
+	rootSegmentBlocksLen = 1
+
+	// expected view of the block in a root sealing segment
+	rootSegmentBlockView = 0
+)
+
+// SealingSegment is the chain segment such that the last block (greatest
+// height) is this snapshot's reference block and the first (least height)
+// is the most recently sealed block as of this snapshot (ie. the block
+// referenced by LatestSeal).
+// In other words, the last block contains a seal for the first block.
+// For instance:
+//   A <- B <- C <- D <- E (seal_A)
+// The above sealing segment's last block (E) has a seal for block A, which is
+// the first block of the sealing segment.
+//
+// It is guaranteed there are at least 2 blocks in a SealingSegment
+// The segment is in ascending height order.
+type SealingSegment struct {
+	// Blocks the chain segment blocks
+	Blocks []*Block
+
+	// Due to decoupling of execution receipts it's possible that blocks from sealing segment will be referring
+	// execution results incorporated in blocks that aren't part of the segment.
+	// ExecutionResults will contain those results.
+	ExecutionResults ExecutionResultList
+}
+
+func (segment *SealingSegment) Highest() *Block {
+	return segment.Blocks[len(segment.Blocks)-1]
+}
+
+func (segment *SealingSegment) Lowest() *Block {
+	return segment.Blocks[0]
+}
+
+var (
+	ErrSegmentMissingSeal        = fmt.Errorf("sealing segment failed sanity check: highest block in segment does not contain seal for lowest")
+	ErrSegmentBlocksWrongLen     = fmt.Errorf("sealing segment failed sanity check: must have atleast 2 blocks")
+	ErrSegmentInvalidBlockHeight = fmt.Errorf("sealing segment failed sanity check: blocks must be in ascending order")
+	ErrSegmentResultLookup       = fmt.Errorf("failed to lookup execution result")
+	ErrInvalidRootSegmentView    = fmt.Errorf("invalid root sealing segment block view")
+)
+
+type SealingSegmentBuilder struct {
+	resultLookup    func(resultID Identifier) (*ExecutionResult, error)
+	includedResults map[Identifier]*ExecutionResult
```

Comment:
I don't see that we use it as map, we only use it for bookkeeping of already seen ERs, we should change it to 
`map[Identifier]struct{}` if we are using this one as set.

Reply:
https://github.com/onflow/flow-go/pull/1533/commits/60a5951058ff9c214eb0b3d85afc5975b50a7721

Reply:
Using struct{}{} https://github.com/onflow/flow-go/pull/1533/commits/ef3887b8355cf2913aa1911d598883538769238a